### ISSUE TITLE
Possible false positive import/namespace error on exported type

### ISSUE THREAD

**Issue by corydeppen** [import-js/eslint-plugin-import#3019]
I'm seeing the following ESLint error when trying to use a namespace import for React Aria Components. I'm unsure whether this could be related to the config in the [default `.eslintrc.cjs` file](https://github.com/remix-run/remix/blob/main/templates/remix/.eslintrc.cjs) of the Remix template or if it might be a false positive being reported. `ButtonProps` is being [exported by the RAC module](https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/Button.tsx#L45), so it's confusing why this error would be shown.

```bash
'ButtonProps' not found in imported namespace 'ReactAria'. eslint(import/namespace)
```

```tsx
import * as ReactAria from "react-aria-components";

// ...

export interface ButtonProps
  extends ReactAria.ButtonProps,
    VariantProps<typeof buttonVariants> {}

// ...
```

The error is not reported when I use the namespace import for exports that aren't types:

```tsx
<ReactAria.Button {...props} />
```

Running `npm run lint` in the [example app](https://stackblitz.com/edit/remix-run-remix-ajyw5v?file=app%2Fcomponents%2Fbutton.tsx) will show the error mentioned above.

Any ideas what might be happening here and whether this is expected or not?

**Comment by ljharb** [comment]
If the import works, I'd expect the plugin to recognize it. What happens if you do `import { ButtonProps } from 'react-aria-components'`?

**Comment by corydeppen** [comment]
There's no error when aliases are used.

```tsx
import {
  Button as RacButton,
  ButtonProps as RacButtonProps,
} from 'react-aria-components';
```

**Comment by ljharb** [comment]
With no aliases, but also with no *?

**Comment by corydeppen** [comment]
No lint errors without aliases or `*` - I've updated the example.

**Comment by corydeppen** [comment]
I'm not sure whether it helps narrow down the issue or not, but I noticed the lint error only seems to be thrown when the namespace import is used to extend a type but not when it's used as an annotation. This can be seen in the [`label` module](https://stackblitz.com/edit/remix-run-remix-ajyw5v?file=app%2Fcomponents%2Flabel.tsx) I just added to the example app.
